### PR TITLE
[ISSUE #8599] Fix send fail with receiving GO_AWAY when rolling update proxy and add channel id in logs

### LINKED ISSUE #8599: [Enhancement] Optimize the graceful shutdown for the proxy remoting protocol server.
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

When the proxy shutdown, gRPC services utilize the HTTP2 GO_AWAY mechanism to gracefully shut down. However, the current implementation of the GO_AWAY mechanism in the remoting protocol still has some unresolved issues, including the lack of server-side logging, repeated channel recreation, and the GO_AWAY only can be received once by client.

### Motivation

Optimize the graceful shutdown for the proxy remoting protocol server.

### Describe the Solution You'd Like

1. add log when proxy write response code  GO_AWAY
2. make client recognize GO_AWAY more then once.

### Describe Alternatives You've Considered

need more discuss

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by qianye1001**
### Which Issue(s) This PR Fixes

Fixes #8599 

1.    Reconnect condition may not be satisfied when rolling update proxy because of the reconnect interval 60 seconds
2.   Sending retry request without extFields may fail

### Brief Description

1.  Fix send fail with receiving GO_AWAY when rolling update proxy
3. Add channel id in logs
4. Send retry request with extFields when receiving GO_AWAY

### CODE REVIEW

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java [hunk 1]
```diff
@@ -819,21 +816,23 @@ public CompletableFuture<ResponseFuture> invokeImpl(final Channel channel, final
...
                             LOGGER.error("Channel {} reconnect error", channelWrapper0, t);
                         }
                         return channelWrapper0;
                     });
-                    if (channelWrapper != null) {
+                    if (channelWrapper != null && !channelWrapper.isWrapperOf(channel)) {
                         if (nettyClientConfig.isEnableTransparentRetry()) {
                             RemotingCommand retryRequest = RemotingCommand.createRequestCommand(request.getCode(), request.readCustomHeader());
                             retryRequest.setBody(request.getBody());
+                            retryRequest.setExtFields(request.getExtFields());
```
Comment (lizhimins) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingClient.java:835]: ...

**Review by lizhimins — DISMISSED** [review]

**Comment by qianye1001** [comment]
@drpmma

**Review by drpmma — APPROVED** [review]
LGTM

**Review by lollipopjin — APPROVED** [review]
LGTM

**Review by lizhimins — APPROVED** [review]

**Pull request merged**